en/24/04.md

```diff
@@ -0,0 +1,335 @@
+                self.zombies(&zombie_id).update(|my_zombie| my_zombie.level += 1);
+            }
+
+            #[only_owner]
+            #[endpoint]
+            fn withdraw(&self) {
+            let caller_address = self.blockchain().get_caller();
+            let collected_fees = self.collected_fees().get();
+            self.send().direct_egld(&caller_address, &collected_fees);
+            self.collected_fees().clear();
+            }
+        }
+    answer: >
+      multiversx_sc::imports!();
+      
+      use crate::{storage, zombie::Zombie, zombiefactory, zombiefeeding, zombiehelper};
+
+      #[multiversx_sc::module]
+      pub trait ZombieAttack:
+          storage::Storage + zombie_feeding::ZombieFeeding + zombie_factory::ZombieFactory + zombie_helper::ZombieHelper
+      {
+          fn rand_mod(&self, modulus: u8) -> u8 {
+              let mut rand_source = RandomnessSource::new();
+              rand_source.next_u8() % modulus
+          }
+      }
+---
+
+Great! Now let's figure out the battle logic.
+
+All good games require some level of randomness. So how do we generate random numbers in Rust?
+
+If you remember we talked about `RandomnessSource` in somewhere in the first lessons.
+
+Due to the nature of the environemnt, nodes must all have the same "random" generator to be able to reach consensus. This is solved by using Golang's standard seeded random number generator, directly in the VM: https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/math/rand/
+
+The VM function mBufferSetRandom uses this library, seeded with the concatenation of:
+
+- previous block random seed
+- current block random seed
+- tx hash
+  
+We're not going to go into details about how exactly the Golang library uses the seed or how it generates said random numbers, as that's not the purpose of this tutorial.
+
+The `ManagedBuffer` type has two methods you can use for this:
+
+- `fn new_random(nr_bytes: usize) -> Self`, which creates a new `ManagedBuffer` of nr_bytes random bytes
+- `fn set_random(&mut self, nr_bytes: usize)`, which sets an already existing buffer to random bytes
+
+For convenience, a wrapper over these methods was created, namely the `RandomnessSource` struct, which contains methods for generating a random number for all base Rust unsigned numerical types, and a method for generating random bytes.
+
+## Put it to the test
+
+Let's implement a random number function we can use to determine the outcome of our battles.
+
+1. Create a function called `rand_mod` (random-modulus). THis function will not be available outside the contract and will have a parameter `u8` named `modulus`, and `returns` a `u8`.
+
+2. The function should first declare a new mutable variable `rand_source` and give it as value `RandomnessSource::new()`;
+
+3. Finally, it should return `rand_source.next_usize() % modulus` making sure the returned number is always smaller than the value of `modulus`
```

Comment:
Please tell them how to make sure the returned number is always smaller than the value of `modulus`. Refer to where this is explained.

Reply:
it's a modulus operation.. the operation itself gives you a number smaller than the operand..
Seems pretty logic for me..